File: packages/python/qsu/os/getUsedRamSize.py
from ..format.fileSizeFormat import fileSizeFormat
from ._system import memorySize


def getUsedRamSize() -> str:
	memory = memorySize()

	if memory is None:
		raise RuntimeError('Failed to read the size of the physical memory')

	total, free = memory

	return fileSizeFormat(max(total - free, 0), 0, True)

File: packages/javascript/lib/node/os/getUsedRamSize.ts
import { freemem, totalmem } from 'node:os';
import { fileSizeFormat } from '../../format/fileSizeFormat.js';

export function getUsedRamSize(): string {
	return fileSizeFormat(Math.max(totalmem() - freemem(), 0), 0, true);
}
